src/components/homepage/EditorHeroSection/EditorHeroSection.jsx

```diff
@@ -0,0 +1,119 @@
+import { CardContainer } from "components/CardContainer"
+import { FormContext, FormError, FormTitle } from "components/Form"
+import FormField from "components/FormField"
+import FormFieldMedia from "components/FormFieldMedia"
+import { HeroDropdown, HeroHighlight } from "components/homepage/HeroSection"
+import _ from "lodash"
+import PropTypes from "prop-types"
+import React, { useEffect } from "react"
+import { useFormContext } from "react-hook-form"
+
+import elementStyles from "styles/isomer-cms/Elements.module.scss"
+
+const HeroComponent = ({ heroType, fieldId }) => {
+  if (heroType === "highlights") return <HeroHighlight fieldId={fieldId} />
+  if (heroType === "dropdown")
+    return <HeroDropdown fieldId={`${fieldId}.dropdown`} />
+  return null
+}
+
+export const EditorHeroSection = ({
+  fieldId, //  This fieldId refers to sections.0.hero, and it's a string reference id to the object in useForm
+}) => {
+  const {
+    register,
+    unregister,
+    formState: { errors },
+    setValue,
+    watch,
+  } = useFormContext()
+  const sectionErrors = _.get(errors, fieldId)
+  const heroType = watch(`${fieldId}.heroType`)
```

Comment:
could i clarify why we aren't passing this in as props but instead chose to have an implicit dependency through observing the `formState` returned from `useFormContext`?